core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -404,13 +404,13 @@ final class KafkaMetadataLog private (
       return false
 
     var didClean = false
-    snapshots.keys.toSeq.sliding(2).toSeq.takeWhile {
+    snapshots.keys.toSeq.sliding(2).foreach {
       case Seq(snapshot: OffsetAndEpoch, nextSnapshot: OffsetAndEpoch) =>
         if (predicate(snapshot) && deleteBeforeSnapshot(nextSnapshot)) {
           didClean = true
           true
```

Comment:
This "true" is not needed now, right?

Reply:
Yeah, this looks weird. Also, early returns within lambdas rely on exceptions for control flow. I bet the issue from before is that `toSeq` was causing `toStream` to be invoked (the Scala Stream is is an odd collection). @mumrah If you do `toBuffer`, I suspect it would work as expected.